infinite_recyclerview/src/main/java/com/az/infinite_recyclerview/InfiniteViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package com.az.infinite_recyclerview
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+abstract class InfiniteViewModel<ITEM : Any> : ViewModel() {
+    protected val _items = MutableLiveData<List<ITEM?>>()
+    val items: LiveData<List<ITEM?>> = _items
+
+    abstract fun hasNextPage(): Boolean
+
+    fun loadMore() {
+        setItemLoadingView(true)
+        getItems()
+    }
+
+    protected abstract fun getItems()
+
+    private fun setItemLoadingView(b: Boolean) {
+        val list = items.value
+        val nullPost: ITEM? = null
+        if (!list.isNullOrEmpty()) {
+            if (b) {
+                _items.value = list.plus(nullPost)
+            } else {
+                if (list[list.size - 1] == null) {
+                    _items.value = list.filterIndexed { index, _ ->
+                        index < list.size - 1
+                    }
+                }
+            }
+        }
```

Comment:
```suggestion
       if (list.isNullOrEmpty()) {
           return
       }
       
       if (b) {
           _items.value = list.plus(nullPost)
           return
       }
       
       if (list[list.size - 1] != null) {
           return
       }
       
       _items.value = list.filterIndexed { index, _ ->
           index < list.size -1
       }
      
```

if - else의 뎁스를 늘리기 보다는 위 형식을 추천합니다.